sklearn_numba_dpex/kmeans/kernels/_base_kmeans_kernel_funcs.py

```diff
@@ -6,6 +6,7 @@
 def _make_initialize_window_kernel_funcs(
     n_clusters,
     n_features,
+    load_centroid_window_half_l2_norm,
```

Comment:
The position of the word "window" makes it hard to understand the meaning of this flag.

I think this should be renamed to `initialize_centroid_half_l2_norm_window` or even `initialize_window_of_centroids_half_l2_norms`.

Also, let's move this param as the last param to make it possible to pass it as a named param to improve readability of calls to this function.